third_party/move/move-core/types/src/value.rs

```diff
@@ -479,6 +523,22 @@ impl MoveStructLayout {
         Self::WithVariants(variants)
     }
 
+    /// Determines whether the layout is serialization compatible with the other layout
+    /// (that is, any value serialized with this layout can be deserialized by the other).
+    pub fn is_compatible_with(&self, other: &Self) -> bool {
+        use MoveStructLayout::*;
+        match (self, other) {
+            (RuntimeVariants(variants1), RuntimeVariants(variants2)) => {
+                variants1.len() <= variants2.len()
+                    && variants1.iter().zip(variants2).all(|(fields1, fields2)| {
+                        MoveTypeLayout::is_compatible_with_slice(fields1, fields2)
+                    })
+            },
+            // All other cases require equality
+            (s1, s2) => s1 == s2,
```

Comment:
This is not correct? If you have a struct that stores an enum inside, you would compare it by equality which will not work I believe, you have to check that field layouts are compatible